nixos/modules/services/mail/goeland.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.goeland;
+  tomlFormat = pkgs.formats.toml { };
+in
+{
+  options.services.goeland = {
```

Comment:
It looks like you should include some sort of option to have the user securely provide a password for the email account. Taking a real quick glance at the source code it looks like the application accepts environment variables so you could add a `passwordFile` that provides this as an environment variable.

Reply:
Great! I didn't catch that, thanks. From here you have 3 options if the user sets the `password` option instead of `password_file`:
- do nothing
- add a warning so the user is reminded it is a really bad idea to add passwords to the nix store
- add an assertion to prevent the user from setting a password insisting they use `password_file` instead

Reply:
I'm having a hard time finding the standard way to warn for this kind of thing. Could you enlighten me please?

Reply:
https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/development/assertions.section.md

Let me know if you need any clarification.

Reply:
Thank you!
Done: https://github.com/NixOS/nixpkgs/pull/206099/files#diff-b130a3abb4626d90fd8792f2a93102b9c99f841b19787b1be01146b57c971ed7R64